homeassistant/components/hp_ilo/sensor.py

```diff
@@ -62,6 +69,8 @@
                         ),
                         vol.Optional(CONF_UNIT_OF_MEASUREMENT): cv.string,
                         vol.Optional(CONF_VALUE_TEMPLATE): cv.template,
+                        vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASSES_SCHEMA,
+                        vol.Optional(CONF_STATE_CLASS): STATE_CLASSES_SCHEMA,
```

Comment:
This is an integration that connects to a device or service. As per ADR-0010.

For integrations that connect to devices or services, we no longer accept new YAML configuration or changes.

This integration needs to be refactored first to use a configuration flow and config entries first.

More information about this can be found in Architecture Decision Record:

- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

See our developer documentation on how to get started creating a configuration flow for this integration:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

Reply:
@sibalzer FWIW, I started such an config flow, including auto-discovery  (but couldn't find the time to wrap it up, i.e. to migrate old config entries and to configure which sensor types shoudl automatically be generated). Feel free to copy from there: https://github.com/chkuendig/hass-hp_ilo-beta